File: src/main/java/com/easy/zssn/repository/SurvivorRepository.java
package com.easy.zssn.repository;

import org.springframework.stereotype.Repository;

import com.easy.zssn.model.Survivor;

import org.springframework.data.jpa.repository.JpaRepository;


@Repository
public interface SurvivorRepository extends JpaRepository<Survivor, Integer> {
    
}

File: src/main/kotlin/com/eduardo/rdguez/zssn/repository/SurvivorRepository.kt
package com.eduardo.rdguez.zssn.repository

import com.eduardo.rdguez.zssn.domain.Survivor
import java.util.*
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface SurvivorRepository : JpaRepository<Survivor, Long> {

  fun findAllByIsInfectedTrue(): List<Survivor>
  fun findByIdAndIsInfectedFalse(id: Long): Optional<Survivor>
  fun countByIsInfectedFalse(): Long
  fun countByIsInfectedTrue(): Long

}